### PR TITLE
feat: use non-zero default sender for l1tol2 messages

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Uses a non-zero default sender (`CALLER`) for L1 => L2 messages. Closes a potential security risk because some people might make the assumption (rightfully so) that one cannot set `CALLER` to be the zero address. 

**Metadata**
- Fixes ENG-1444

### CODE REVIEW

**Comment by smartcontracts** [comment]
To be discussed: is this default sender OK? Or should we use a random address instead?

File: integration-tests/test/basic-l1-l2-communication.spec.ts [hunk 1]
```diff
@@ -97,6 +96,25 @@ describe('Basic L1<>L2 Communication', async () => {
       expect((await L2SimpleStorage.totalCount()).toNumber()).to.equal(1)
     })
 
+    it('should be able to send L1 -> L2 directly via enqueue', async () => {
```
Comment (tynes) [inline integration-tests/test/basic-l1-l2-communication.spec.ts:99]: This name should be a bit more descriptive that it covers the fact `CALLER` is the default value so its easier to remember that is covered via tests

**Comment by maurelian** [comment]
> To be discussed: is this default sender OK? Or should we use a random address instead?

Perhaps something inspired by the pattern in [EIP-1967](https://github.com/ethereum/EIPs/blob/bbf07a122d0ba90d2cf42911f1d00990e89cc96a/EIPS/eip-1967.md#admin-address): 
`bytes20(uint256(keccak256('default.l1Origin')) - 1)`

**Comment by tynes** [comment]
Tests are failing, unclear as to why

**Comment by tynes** [comment]
This should be specified and merged asap as it is a non backwards compatible change and will require a restart of a testnet

**Comment by smartcontracts** [comment]
Closing in favor of #1493

**Pull request closed without merging**